perceval/backends/gerrit.py

```diff
@@ -0,0 +1,297 @@
+        self._push_cache_queue(raw_data)
+        self._flush_cache_queue()
+        reviews = self._parse_reviews(raw_data)
+        self.last_item = self.client.get_next_item(self.last_item, reviews[-1])
+        logging.info("Received %i reviews in %.2fs" % (len(reviews),
+                                                       time()-task_init))
+        return reviews
+
+    def _parse_reviews(self, raw_data):
+        # Join isolated reviews in JSON in array for parsing
+        items_raw = "[" + raw_data.replace("\n", ",") + "]"
+        items_raw = items_raw.replace(",]", "]")
+        items = json.loads(items_raw)
+        reviews = []
+
+        for item in items:
+            if 'rowCount' in item.keys():
+                # stats data, not a review
+                self.number_results = item['rowCount']
+            elif 'project' in item.keys():
+                reviews.append(item)
+
+        return reviews
+
+
+class GerritClient():
+
+    def __init__(self, repository, user, nreviews):
+        self.gerrit_user = user
+        self.nreviews = nreviews
+        self.repository = repository
+        self.project = None
+        self.version = None
+        self.gerrit_cmd = "ssh -p 29418 %s@%s" % (self.gerrit_user,
+                                                  self.repository)
+        self.gerrit_cmd += " gerrit "
+
+    def _get_version(self):
+
+        if self.version:
+            return self.version
+
+        cmd = self.gerrit_cmd + " version "
+
+        logging.debug("Getting version: %s" % (cmd))
+        raw_data = subprocess.check_output(cmd, shell=True)
+        raw_data = str(raw_data, "UTF-8")
+        logging.debug("Gerrit version: %s" % (raw_data))
+
+        # output: gerrit version 2.10-rc1-988-g333a9dd
+        m = re.match("gerrit version (\d+)\.(\d+).*", raw_data)
+
+        if not m:
+            cause = "Invalid gerrit version %s" % raw_data
+            raise BackendError(cause=cause)
+
+        try:
+            mayor = int(m.group(1))
+            minor = int(m.group(2))
+        except:
+            cause = "Gerrit client could not determine the server version."
+            raise BackendError(cause=cause)
+
+        self.version = [mayor, minor]
+        return self.version
+
+    def _get_gerrit_cmd(self, last_item):
+
+        cmd = self.gerrit_cmd + " query "
+        if self.project:
+            cmd += "project:"+self.project+" "
+        cmd += "limit:" + str(self.nreviews)
+
+        # This does not work for Wikimedia 2.8.1 version
+        cmd += " '(status:open OR status:closed)' "
+
+        cmd += " --all-approvals --comments --format=JSON"
+
+        gerrit_version = self._get_version()
+
+        if last_item is not None:
+            if gerrit_version[0] == 2 and gerrit_version[1] >= 9:
+                cmd += " --start=" + str(last_item)
+            else:
+                cmd += " resume_sortkey:" + last_item
+
+        return cmd
+
+    def get_items(self, last_item):
```

Comment:
I like items too but I'm afraid it's impossible since we will have clients that won't return just items. Please check the code from Bugzilla backend. We have `buglist`, `bugs`, `bug_activity`. In GitHub backend we will have issues, pull_requests, even users!